packages/blob/src/client.ts

```diff
@@ -232,8 +236,12 @@ export async function handleUpload({
   const type = body.type;
   switch (type) {
     case 'blob.generate-client-token': {
-      const { pathname, callbackUrl } = body.payload;
-      const payload = await onBeforeGenerateToken(pathname);
+      const { pathname, callbackUrl, clientPayload } = body.payload;
+      const payload = await onBeforeGenerateToken(pathname, clientPayload);
+      // TODO @Fabio: the next line `?? null` previously, which always made sure that tokenPayload was `null`
```

Comment:
cc @correttojs the previous code was : `metadata: payload.metadata ?? null` which ensured the JSON was always `null`, while other types suggest metadata is always either undefined or a string, any reason why?

Reply:
I remember there was something with the metadata serialization. not sure if the `undefined` would cause any issue while serializing/deserializing the client token.
We can add a generateToken unit test passing an empty payload eventually

Reply:
Added a unit test, fine